src/submission/schema/schema-manager.ts

```diff
@@ -268,20 +282,77 @@ namespace MigrationManager {
       newSchemaVersion,
     );
 
-    await checkDonorDocuments(migration, newTargetSchema);
+    const migration1 = await checkDonorDocuments(migration, newTargetSchema);
+
+    await revalidateOpenSubmissionsWithNewSchema(migration1, newTargetSchema, migration.dryRun);
 
     // close migration
     const updatedMigration = await migrationRepo.getById(migrationId);
     const migrationToClose = _.cloneDeep(updatedMigration) as DictionaryMigration;
+
     if (!migrationToClose) {
       throw new Error('where did the migration go? expected migration not found');
     }
+
     migrationToClose.state = 'CLOSED';
     migrationToClose.stage = 'COMPLETED';
     const closedMigration = await migrationRepo.update(migrationToClose);
     return closedMigration;
   };
 
+  const revalidateOpenSubmissionsWithNewSchema = async (
+    migration: DictionaryMigration,
+    newSchema: SchemasDictionary,
+    dryRun: boolean,
+  ) => {
+    const submissions = await submissionService.operations.findActiveClinicalSubmissions();
+
+    for (const sub of submissions) {
+      if (
+        sub.state == SUBMISSION_STATE.INVALID ||
+        sub.state == SUBMISSION_STATE.INVALID_BY_MIGRATION
+      ) {
+        continue;
+      }
+
+      const checkedSubmission = migration.checkedSubmissions.find(x => {
+        x._id = sub._id;
+      });
+
+      if (checkedSubmission) {
+        continue;
+      }
+
+      migration.checkedSubmissions.push({
+        programId: sub.programId,
+        id: sub._id,
+      });
+
+      const command: RevalidateClinicalSubmissionCommand = {
+        programId: sub.programId,
+        migrationId: migration._id as string,
+      };
+
+      const result = await submissionService.operations.revalidateClinicalSubmission(
+        command,
+        newSchema,
+        dryRun,
+      );
+      if (!result.submission) {
+        continue;
+      }
+
+      if (result.submission.state == 'INVALID_BY_MIGRATION') {
+        migration.invalidSubmissions.push({
+          programId: sub.programId,
+          id: sub._id,
+        });
+      }
+
+      migration = await migrationRepo.update(migration);
```

Comment:
Are we modifying the migration variable that is a method argument? should we instead build this without side effects, returning a new migration object?

Reply:
I need to update the migration document in every loop iteration, and this is a private method it just to break down the code .. im not sure if this is considered a side effect because most of our business logic interacts with a repository class of some kind.. I'm not sure that I understand the problem

Reply:
Being private (used only locally here) means no risk of being called in the future and causing unexpected changes. Good enough for me.